.github/workflows/deployment.yml

```
@@ -0,0 +1,99 @@
+name: 'Deployment to GCP'
+
+on:
+  push:
+    branches:
+      - main
+
+
+jobs:
+  build-infrastructure:
+    name: 'Set up infrastructure'
+    runs-on: ubuntu-latest
+    env:
+      GOOGLE_APPLICATION_CREDENTIALS: /tmp/credentials.json
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@v1
+
+      - name: Set Up Credentials
+        env:
+          SET_UP_CREDENTIALS: ${{ secrets.GOOGLE_CREDENTIALS }}
+        run: printf '%s' "$SET_UP_CREDENTIALS" > "$GOOGLE_APPLICATION_CREDENTIALS"
```

```suggestion
    env:
      GOOGLE_APPLICATION_CREDENTIALS_PATH: /tmp/credentials.json

    defaults:
      run:
        shell: bash

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Set up Terraform
        uses: hashicorp/setup-terraform@v1

      - name: Set Up Credentials
        env:
          SET_UP_CREDENTIALS: ${{ secrets.GOOGLE_CREDENTIALS }}
        run: printf '%s' "$SET_UP_CREDENTIALS" > "$GOOGLE_APPLICATION_CREDENTIALS_PATH"
```
i would call it path or directly use it

```suggestion
    defaults:
      run:
        shell: bash

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Set up Terraform
        uses: hashicorp/setup-terraform@v1

      - name: Set Up Credentials
        env:
          SET_UP_CREDENTIALS: ${{ secrets.GOOGLE_CREDENTIALS }}
        run: printf '%s' "$SET_UP_CREDENTIALS" > "/tmp/credentials.json"
```

---

totally get your point but this is actually a requirement from TF plus gcloud, TF expects to find a path in the env var `GOOGLE_APPLICATION_CREDENTIALS` under this very specific name. Therefore, for the TF steps to work with the env var I need to do it like this 😔